blockly/src/utils/pathfinding/Node.java

```diff
@@ -0,0 +1,61 @@
+package utils.pathfinding;
+
+import core.level.utils.Coordinate;
+import java.util.List;
+import utils.LevelUtils;
+
+/**
+ * A Node represents a point in the pathfinding grid.
+ *
+ * <p>Each node has a coordinate and a list of its walkable neighbors. This class is used in
+ * pathfinding algorithms to represent the state of the search.
+ *
+ * <p>It provides methods to access the coordinate and neighbors of the node.
+ *
+ * @see PathfindingLogic
+ */
+public class Node {
```

Comment:
Das ist einfach nur ein `Tile` (plus Liste betretbarer Nachbarn), oder? 

Ich würde die Nachbar-Berechnung in die Suche auslagern und nicht statisch berechnen und dann kannst Du Dir diese Klasse sparen und direkt `Tile` nehmen?

Alternativ riecht das hier ziemlich nach einem Record?